File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        traversal = []
        self.inOrderTraversalHelper(root, traversal)
        return traversal


    def inOrderTraversalHelper(self, root, traversal):
        if not root:
            return

        self.inOrderTraversalHelper(root.left, traversal)
        traversal.append(root.val)
        self.inOrderTraversalHelper(root.right, traversal)

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> rightVals;
        queue<TreeNode*> q;

        if (root) {
            q.push(root);
        }

        while (q.size() > 0) {
            int qSize = q.size();
            TreeNode* curr;
            for (int i = 0; i < qSize; i++) {
                curr = q.front();
                q.pop();

                if (curr->left) {
                    q.push(curr->left);
                }

                if (curr->right) {
                    q.push(curr->right);
                }
            } 

            rightVals.push_back(curr->val);
        }

        return rightVals;
    }
};
